File: csharp/src/DivisorAndWord.cs
namespace csharp;

public class DivisorAndWord
{
    private readonly int _divisor;
    private readonly string _word;

    public DivisorAndWord(int divisor, string word)
    {
        _divisor = divisor;
        _word = word;
    }

    public string GetWordOrEmpty(int number)
    {
        return number % _divisor == 0 ? _word : "";
    }
}

File: typescript/src/DivisorAndWord.ts
export default class DivisorAndWord {
    private readonly divisor: number;
    private readonly word: string;

    constructor(divisor: number, word: string) {
        this.divisor = divisor;
        this.word = word;
    }

    public getWordOrEmpty(inputNumber: number): string {
        return inputNumber % this.divisor === 0 ? this.word : "";
    }
}
